hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/validation/ExpiryValidatorImpl.java

```diff
@@ -74,15 +77,32 @@ public ExpiryMeta resolveCreationAttempt(
         if (hasCompleteAutoRenewSpec(entityCanSelfFundRenewal, creationMeta)) {
             effectiveExpiry = clampedAdd(context.consensusNow().getEpochSecond(), creationMeta.autoRenewPeriod());
         }
-        // In mono-service this check is done first for topic creation.
         // To maintain same behaviour for differential testing this condition is needed.
         // FUTURE: This condition should be removed after differential testing is done
-        if (isForTopicCreate) {
+        if (functionality == HederaFunctionality.CONSENSUS_CREATE_TOPIC) {
+            // In mono-service this check is done first for topic creation.
             context.attributeValidator().validateExpiry(effectiveExpiry);
             // Even if the effective expiry is valid, we still also require any explicit auto-renew period to be valid
             if (creationMeta.hasAutoRenewPeriod()) {
                 context.attributeValidator().validateAutoRenewPeriod(creationMeta.autoRenewPeriod());
             }
+        } else if (functionality == HederaFunctionality.TOKEN_CREATE) {
+            // In mono-service, INVALID_RENEWAL_PERIOD is thrown for token creation and update
+            // if the autoRenewPeriod is not in range.
+            // It would be more correct to throw AUTO_RENEW_DURATION_NOT_IN_RANGE like the other services do,
+            // but this would break differential testing. So for now, we replicate the mono-service behaviour.
+            try {
```

Comment:
Instead of duplicating this block, can't we extract it to a method?

Reply:
Fixed